botocore/serialize.py

```diff
@@ -518,7 +536,12 @@ def _convert_header_value(self, shape, value):
         if shape.type_name == 'timestamp':
             datetime_obj = parse_to_aware_datetime(value)
             timestamp = calendar.timegm(datetime_obj.utctimetuple())
-            return self._timestamp_rfc822(timestamp)
+            timestamp_format = shape.serialization.get('timestampFormat')
```

Comment:
Similar feedback as above, any reason not to do:
```python
timestamp_format = shape.serialization.get('timestampFormat', 'rfc822')
return self._convert_timestamp_to_str(timestamp, timestamp_format)
```
I think I would prefer this as it seems the precedent is to use `_convert_timestamp_to_str` instead of calling the `_timestamp_<format>` methods directly.

At the very least we should be consistent in the current diff and have this be a similar style class variable like `HEADER_TIMESTAMP_FORMAT = 'rfc822'` or just set the default for the `get` call for both of these edge cases.